lib/SIL/SILType.cpp

```diff
@@ -319,7 +319,8 @@ SILType SILType::getEnumElementType(EnumElementDecl *elt, SILModule &M) const {
   // If the case is indirect, then the payload is boxed.
   if (elt->isIndirect() || elt->getParentEnum()->isIndirect())
     loweredTy = SILType::getPrimitiveObjectType(
-      SILBoxType::get(loweredTy.getSwiftRValueType()));
+      SILBoxType::get(loweredTy.getSwiftRValueType(),
+                      /*immutable*/ false));
```

Comment:
Indirect payloads really are immutable though, aren't they? We have no way to refer to them as L values.